### ISSUE TITLE
Semantics for `Checkbox` should react to the three-state mixed state

### ISSUE THREAD

**Issue by yaakovschectman** [flutter/flutter#110107]
Continuation of #96218. When a three-state `Checkbox` is in the mixed state, the semantic flags for it do not update to reflect this.

Minimal example:

```
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool? isChecked = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: ListView(
        children: [
          CheckboxListTile(
            autofocus: true,
            onChanged: (value) => setState(() {
              isChecked = value;
            }),
            title: const Text('Tristate Value Checked'),
            tristate: true,
            value: isChecked,
          ),
          CheckboxListTile(
            value: isChecked ?? true,
            onChanged: (value) => setState(() {
              isChecked = value ?? false;
            }),
            title: const Text('Flag'),
          )
        ],
      ),
    );
  }
}
```

The tristate `Checkbox` should set the system object state `STATE_SYSTEM_MIXED` when it is in the mixed state. Currently, instead, it can be only in the checked or unchecked states dictated by `STATE_SYSTEM_CHECKED`.

This will require changes to the engine, as well.

**Comment by yaakovschectman** [comment]
Closed by #111032

**Comment by chrisnorman7** [comment]
Hi,
Thank you so much for the work you've put ointo this.

Tristate now works perfectly on Windows. Unfortunately the half-checked state is not read by NVDA when running in Chrome.
